ci/nodeEngineCheck.js

```diff
@@ -76,16 +76,20 @@ class NodeEngineCheck {
     // For each file
     for (const file of files) {
 
-      // Get node version
-      const contentString = await fs.readFile(file, 'utf-8');
-      const contentJson = JSON.parse(contentString);
-      const version = ((contentJson || {}).engines || {}).node;
-      
-      // Add response
-      response.push({
-        file: file,
-        nodeVersion: version
-      });
+      try {
+        // Get node version
+        const contentString = await fs.readFile(file, 'utf-8');
+        const contentJson = JSON.parse(contentString);
+        const version = ((contentJson || {}).engines || {}).node;
+
+        // Add response
+        response.push({
+          file: file,
+          nodeVersion: version
+        });
+      } catch {
+        console.log(`Skipping ${file} due to package.json parsing issues`);
```

Comment:
Not sure about silent fail; if it fails then we may overlook something. For which files did you see it failing?

Reply:
For `eslint-plugin-react/node_modules/resolve/test/resolver/malformed_package_json/package.json` which is malformed specifically for their tests.

Reply:
Got it, I think the better option here is to change the script to exclude certain paths (defined by regex) from parsing.

Reply:
To be honest, I do not like how currently `engine` is being tested. I would propose to use npm's config flag called [`engine-strict`](https://docs.npmjs.com/cli/v7/using-npm/config#engine-strict) to stop CI if any of dependencies does not meet requirements.

Reply:
Unfortunately it seems like `engine-strict` option does not work for npm@6, tested it locally and it does not work at all.

Reply:
engine-strinct doesn't do the same as our script. If engines.node is for example >=12 and a dependency is >=14, then engine-strict will pass, as far as I remember. Our script will reject, because the dependency requires a higher node version. I wrote the script because currently there is no npm option for that.

Reply:
Based on what npm's docs describe if you set `engine-strict` to `true` and try to install a dep that requires Node 14 on OS that has Node 12 set as a default, the installation will fail. I tried it in one PR, but it seems broken on npm 6, on npm 7/8 it works for some, but that would require additional work.

Reply:
I would be happy if we could remove our script and replace it with `engine-strict`, but I'm not convinced yet.

> If set to true, then npm will stubbornly refuse to install (or even consider installing) any package that claims to not be compatible with the current Node.js version.

That means it won't install based on the Node.js version, not based on `engined.node` in `package.json`. The script currently considers the `engined.node` version, not the installed Node version. To achieve the same, it means we have to add an additional Node environment in the CI with the version set in `engined.node`. That means currently for example, we need two Node 12 CI environments; the absolute minimum Node 12 version in `engined.node` and the latest Node 12 (minor/patch) version.